cow_instrument/DetectorInfo.h

```diff
@@ -40,27 +40,45 @@ template <typename InstTree> class DetectorInfo {
 
   bool isMonitor(size_t detectorIndex) const;
 
-  void initL2();
+  double l2(size_t detectorIndex) const;
 
-  void initL1();
+  Eigen::Vector3d position(size_t componentIndex) const;
 
-  double l2(size_t detectorIndex) const;
+  Eigen::Vector3d positionDetector(size_t detectorIndex) const;
```

Comment:
Adding type-safe indices will not make the interface much better, I think. I was thinking more along the lines of having separate objects for separate tasks. I am not sure if that makes sense though, what do you think?